Packages/com.chisel.editor/Chisel/Editor/Editor/MaterialBrowser/ChiselMaterialBrowserUtilities.cs

```diff
@@ -0,0 +1,242 @@
+/* * * * * * * * * * * * * * * * * * * * * *
+Chisel.Editors.ChiselMaterialBrowserCache.cs
+
+License: MIT (https://tldrlegal.com/license/mit-license)
+Author: Daniel Cornelius
+
+* * * * * * * * * * * * * * * * * * * * * */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Chisel.Components;
+using UnityEditor;
+using UnityEngine;
+
+namespace Chisel.Editors
+{
+    internal static class ChiselMaterialBrowserUtilities
+    {
+        // checks a path and returns true/false if a material is ignored or not
+        public static bool IsValidEntry( ChiselMaterialBrowserTile tile )
+        {
+            // these are here to clean things up a little bit and make it easier to read
+
+            bool PathContains( string path )
+            {
+                return tile.path.ToLower().Contains( path );
+            }
+
+            // checks for any shaders we want to exclude
+            bool HasInvalidShader()
+            {
+                string shader = tile.shaderName.ToLower();
+
+                string[] excludedShaders = new string[]
+                {
+                        "skybox/"
+                };
+
+                return shader.Contains( excludedShaders[0] );
+            }
+
+            string chiselPath = "packages/com.chisel.components/package resources/";
+
+            string[] ignoredEntries = new string[]
+            {
+                    "packages/com.unity.searcher/",    // 0, we ignore this to get rid of the built-in font materials
+                    "packages/com.unity.entities/",    // 1, we ignore this to get rid of the entities materials
+                    $"{chiselPath}preview materials/", // 2, these are tool textures, so we are ignoring them
+            };
```

Comment:
Again, this can be a static readonly thing outside the method. This will allocate garbage everytime the method is called ..

Reply:
This is fixed in next commit.